independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -131,7 +135,8 @@ public static boolean isResourcePresentAtRuntime(String resourcePath) {
         PLATFORM_CLASS_LOADER = cl;
     }
 
-    private boolean closed;
+    private volatile boolean closing;
+    private volatile boolean closed;
```

Comment:
A single variable representing a state will be better. I.e. in this case something like `state` with possible values `OPEN`, `CLOSING`, `CLOSED`, a type could be `int` or `byte`.

Reply:
Yeah, I was expecting that you would say that :).